src/diffusers/models/unets/unet_2d_blocks.py

```diff
@@ -1042,10 +1067,32 @@ def forward(
         output_states = ()
 
         for resnet, attn in zip(self.resnets, self.attentions):
-            cross_attention_kwargs.update({"scale": lora_scale})
-            hidden_states = resnet(hidden_states, temb, scale=lora_scale)
-            hidden_states = attn(hidden_states, **cross_attention_kwargs)
-            output_states = output_states + (hidden_states,)
+            if self.training and self.gradient_checkpointing:
+
+                def create_custom_forward(module, return_dict=None):
+                    def custom_forward(*inputs):
+                        if return_dict is not None:
+                            return module(*inputs, return_dict=return_dict)
+                        else:
+                            return module(*inputs)
+
+                    return custom_forward
+
+                ckpt_kwargs: Dict[str, Any] = {"use_reentrant": False} if is_torch_version(">=", "1.11.0") else {}
+                cross_attention_kwargs.update({"scale": lora_scale})
+                hidden_states = torch.utils.checkpoint.checkpoint(
```

Comment:
I don't think `scale=lora_scale` argument is being passed down to `resnet()` when checkpointing? did I miss anything? 
cc @sayakpaul